frontend/src/Routes/User/UserProfile/index.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useParams } from "react-router-dom";
+import { Row, Col } from "antd";
+
+import { connect, storeType } from '../../../Store';
+import { getProfileInfoByUserId } from '../../../Api/User';
+
+
+const BoxedList = ({ title, keyValueList }) => {
+  const styles = {
+    key: {
+      fontWeight: "bold",
+    },
+    title: {
+      fontWeight: "bold",
+      fontSize: "%120"
+    }
+  }
+  return (
+  <Col>
+    <Row style={styles.title}>{title}</Row>
+    {keyValueList.map( ({key, value}) => (
+      <Row>
+        <Col span={10} style={styles.key}>{key}</Col>
+        <Col span={14}>{value}</Col>
+      </Row>
+    ))}
+  </Col>
+)};
+
+const UserProfile = ({ store: { userId } }) => {
+  const params = useParams();
+  const paramUserId = params.userId;
+  if (paramUserId) {
+    userId = paramUserId;
+  }
+  const [profile, setProfile] = useState();
+  useEffect(() => {
+    if (userId) {
+      getProfileInfoByUserId(userId)
+        .then((response) => {
+          const { data = {} } = response.data || {};
+          setProfile(data);
+        })
+        .catch(console.log);
+    }
+  }, [userId]);
+
+  const { email, firstName, grades, languages, lastName, progressLevels } = profile || {};
+
+  const languagesBoxFriendly = (languages || []).map((lang, i) => (
+    [lang, grades[i], progressLevels[i]]
```

Comment:
here, instead of using grades[i] directly (which is shown as a number 1-6) you should use numGradeToStrGrade function in Helpers/grades.js  which converts that number to a human readable form.